src/ado/reprun.ado

```diff
@@ -255,22 +262,52 @@ end
         * Not part of a multiline line
         else {
 
-          *Reset the last line local
+          * Reset the last line locals
           local last_line = ""
           local line_command = "OTHER"
           local dofile ""
           local doflag 0
-          foreach w in `macval(line)' {
-            get_command, word("`w'")
+          local looptype ""
+          local loopflag 0
+
+          // Sanitize that string! -- see d17586d873a978987f34ba2fe536a311107ea58b for more regex
+          local theline = `"`macval(line)'"'
+          while regexm(`"`macval(theline)'"',"[\*]") {
+            local theline = regexr(`"`macval(theline)'"',"[\*]","")
+          }
+          while regexm(`"`macval(theline)'"',"\[//]"){
+            local theline = regexr(`"`macval(theline)'"',"\[//]","")
+          }
+
+          // Identify all commands in line
+          foreach w in `macval(theline)' {
+            cap get_command, word(`"`w'"')
+
             if `doflag' == 1 local dofile = "`w'"
+            if `loopflag' == 1 local looptype = "`w'"
+
+            * Dofiles
             if "`r(command)'" == "do" | "`r(command)'" == "run" {
               local doflag = 1
             }
             else local doflag 0
+
+            * Loops
+            if "`r(command)'" == "foreach" | "`r(command)'" == "forvalues" {
+              local loopflag = 1
+            }
+            else local loopflag 0
```

Comment:
@kbjarkefur Note the new handling for loops -- this should fix the reporting issues.